graphql-ballerina/validator_visitor.bal

```diff
@@ -276,7 +276,7 @@ class ValidatorVisitor {
             __Type ofType = getOfType(schemaType);
             if (fragmentOnType != ofType) {
                 string message = getFragmetCannotSpreadError(fragmentNode, fragment.name, ofType);
-                ErrorDetail errorDetail = getErrorDetailRecord(message, fragment.location);
+                ErrorDetail errorDetail = getErrorDetailRecord(message, fragment?.spreadLocation ?: fragment.location);
```

Comment:
Here, (also in the above scenario), we know that the selection has a `spreadLocation` value right? In that case, we can directly cast it and use it, instead of the condition. WDYT?
```suggestion
                ErrorDetail errorDetail = getErrorDetailRecord(message, <Location>fragment?.spreadLocation);
```